### PR TITLE
Update CAA Record Protobuf

### PR DESCRIPTION
**Pull request by kumarde**
Update to move validation result on the certificate level in addition to the domain level.

### CODE REVIEW

**Comment by zakird** [comment]
@kumarde please update this so that it doesn't conflict with master.

File: proto/caa.proto [hunk 1]
```diff
@@ -1,39 +1,50 @@
...
-    CAA_STATUS_DNS_ERROR_BLACKLIST = 7;
-    CAA_STATUS_DNS_ERROR_NO_OUTPUT = 8;
-    CAA_STATUS_DNS_ERROR_NO_ANSWER = 9;
-    CAA_STATUS_DNS_ERROR_ILLEGAL_INPUT = 10;
-    CAA_STATUS_DNS_ERROR_TIMEOUT = 11;
-    CAA_STATUS_DNS_ERROR_ITER_TIMEOUT = 12;
-    CAA_STATUS_DNS_ERROR_TEMPORARY = 13;
-    CAA_STATUS_DNS_ERROR_TRUNCATED = 14;
-}
+    CAA_STATUS_DNS_SUCCESS = 4;
```
Comment (zakird) [inline proto/caa.proto]: Why would this be a thing? if DNS lookup was successful, then shouldn't this be a `CAA_STATUS_VALIDATION_SUCCESS` or `CAA_STATUS_VALIDATION_FAIL`?

File: proto/caa.proto [hunk 2]
```diff
@@ -1,39 +1,50 @@
...
-    CAA_STATUS_RESERVED = 0;
+enum CAAResult {
+    CAA_RESULT_RESERVED = 0;
+    CAA_RESULT_VALIDATION_SUCCESS = 1;
+    CAA_RESULT_VALIDATION_FAIL = 2;
+    CAA_RESULT_VALIDATION_SKIPPED = 3;
+    CAA_RESULT_VALIDATION_NOT_REQUIRED = 4;
+}
+
+enum CAAStatusDNS {
```
Comment (zakird) [inline proto/caa.proto]: Why does this have DNS appended to the end now? These are't all DNS results. Also, even if these _were_ DNS, why don't you put DNS in a normal location in the name?

Comment (zakird) [reply proto/caa.proto]: Having `VALIDATION` in all of these makes me believe that this Enum is either incorrectly named or all the values in the Enum are incorrectly named.

File: proto/caa.proto (hunk 1 above)
Comment (kumarde) [reply proto/caa.proto]: Yeah you're right, it's not a thing.

File: proto/caa.proto (hunk 2 above)
Comment (kumarde) [reply proto/caa.proto]: Yeah the ENUM is poorly named. Fixing

**Pull request merged**